uui-components/src/buttons/IconButton.tsx

```diff
@@ -1,19 +1,37 @@
 import * as React from 'react';
-import { IDropdownToggler, Icon } from '@epam/uui-core';
+import { IDropdownToggler, Icon, IClickable, IAnalyticableClick, IHasTabIndex, IDisableable, IHasCX, ICanRedirect } from '@epam/uui-core';
+import { AnchorNavigationProps, ButtonNavigationProps, Clickable, HrefNavigationProps, LinkButtonNavigationProps } from '../widgets';
 import { IconContainer } from '../layout';
-import { Clickable, ClickableComponentProps } from '../widgets';
 import css from './IconButton.module.scss';
 
-export type IconButtonProps = ClickableComponentProps & IDropdownToggler & {
+export type UnionIconButtonNavigationProps = HrefNavigationProps | LinkButtonNavigationProps | ButtonNavigationProps | AnchorNavigationProps;
+
+export type IconButtonRawProps =
+    | (React.AnchorHTMLAttributes<HTMLAnchorElement>
+    | React.ButtonHTMLAttributes<HTMLButtonElement>)
+    & Record<`data-${string}`, string>;
+
+export type IconButtonProps = IClickable & IAnalyticableClick & IHasTabIndex & IDisableable & IHasCX
+& Omit<ICanRedirect, 'href' | 'link'> & IDropdownToggler & UnionIconButtonNavigationProps & {
```

Comment:
why do we omit here 'href' and 'link'?

Reply:
We have discriminated union, in my point of view in this case - 'href' | 'link' redundant

Reply:
but why we shouldn't do the same for all other components?